interface.go

```diff
@@ -22,6 +22,12 @@ type Event interface {
 type Repository interface {
 	// Gets the Events which should follow on from the specified channel and event id. This method may be called
 	// from different goroutines, so it must be safe for concurrent access.
+	//
+	// It is important for the Repository to close the channel after all the necessary events have been
+	// written to it. The stream will not be able to proceed to any new events until it has finished consuming
+	// the channel that was returned by Replay.
```

Comment:
This isn't exactly a new requirement, but the behavior has changed a bit.

Previously, if we might need to replay events, Server would start a worker goroutine which would consume events from the channel that was returned by the Repository, and send them one by one onto the subscriber's channel. If the Repository was very slow or did not close the channel, this worker goroutine would be blocked and might linger around forever, but it would not block the subscriber from receiving new events that were published. That was a mixed blessing because it also meant that there was no defined ordering between the replayed events and the new events— they might be interleaved, which is contrary to the purpose of the Replay mechanism.

Now, until the Repository closes the channel, it will be blocking the subscriber goroutine (`Server.Handle()`). The subscriber will not be able to get new events during that time, but I think that is correct because the "replayed" events are by definition stuff that comes before the new events. The `Server.run()` goroutine will _not_ be blocked. And if the client gets tired of waiting and closes the socket, then the subscriber goroutine will stop waiting too.